### PR TITLE
Remove jQuery AJAX from the project page

### PR DESCRIPTION
**Pull request by yardenshoham**
Removed all jQuery AJAX calls and replaced with our fetch wrapper.

Tested the following functionalities and they work as before:
- column creation
- column deletion
- issue movement between columns
- column reordering
- column edit
- default column changing

# Demo using `fetch` instead of jQuery AJAX
demo

### CODE REVIEW

File: web_src/js/features/repo-projects.js [hunk 1]
```diff
@@ -2,31 +2,35 @@ import $ from 'jquery';
...
+    });
+
+    if (!response.ok) {
+      throw new Error('Failed to create new column');
+    }
+  } catch (error) {
+    console.error(error);
+  } finally {
     columnTitle.closest('form').removeClass('dirty');
     window.location.reload();
```
Comment (silverwind) [inline web_src/js/features/repo-projects.js:25]: I wouldn't reload the page in error case. This "Failed to create new column" error sounds like one that should show a toast.

File: web_src/js/features/repo-projects.js [hunk 2]
```diff
@@ -38,18 +42,18 @@ function moveIssue({item, from, to, oldIndex}) {
...
-    },
-  });
+  try {
+    const response = await POST(`${to.getAttribute('data-url')}/move`, {
+      data: columnSorting,
+    });
+
+    if (!response.ok) {
+      throw new Error('Failed to move issue');
+    }
```
Comment (silverwind) [inline web_src/js/features/repo-projects.js]: Same here, show it.

File: web_src/js/features/repo-projects.js (hunk 1 above)
Comment (yardenshoham) [reply web_src/js/features/repo-projects.js:25]: This PR isn't meant to change any functionality, it's out of scope. This is purely to remove jQuery AJAX.

Comment (silverwind) [reply web_src/js/features/repo-projects.js:25]: Yeah I understand it. We can keep this broken behaviour.

But I doubt the benefit of logging an error and then reloading the page. Browser console is usually set up so it clears on reload. So maybe just don't log it at all if that was the previous behaviour.

Comment (yardenshoham) [reply web_src/js/features/repo-projects.js:25]: https://github.com/go-gitea/gitea/pull/29814/commits/0520d2f6c5c1b2c87fe0b4abfec9f1da1acf76cd

**Review by silverwind — APPROVED** [review]

File: web_src/js/features/repo-projects.js [hunk 3]
```diff
@@ -190,7 +182,7 @@ export function initRepoProject() {
     if (!columnTitle.val()) {
       return;
     }
-    const url = $(this).data('url');
+    const url = e.target.getAttribute('data-url');
```
Comment (silverwind) [inline web_src/js/features/repo-projects.js:185]: Interestin, I though we had already forbidden `.data()`, but apparently not.

Comment (yardenshoham) [reply web_src/js/features/repo-projects.js:185]: It's possible it's forbidden in plain JS dataset but not in jQuery

Comment (silverwind) [reply web_src/js/features/repo-projects.js:185]: Ah yes, that might be the case. Should be an easy refactor.

File: web_src/js/features/repo-projects.js [hunk 4]
```diff
@@ -38,18 +42,18 @@ function moveIssue({item, from, to, oldIndex}) {
...
-      from.insertBefore(item, from.children[oldIndex]);
-    },
-  });
+  try {
+    const response = await POST(`${to.getAttribute('data-url')}/move`, {
+      data: columnSorting,
+    });
+
+    if (!response.ok) {
+      throw new Error('Failed to move issue');
```
Comment (delvh) [inline web_src/js/features/repo-projects.js]: I'd like some more info in this message such as
```suggestion
      throw new Error(`Failed to move issue - received status: ${response.status} ${response.statusText}`);
```

File: web_src/js/features/repo-projects.js [hunk 5]
```diff
@@ -2,31 +2,35 @@ import $ from 'jquery';
...
+  try {
+    const response = await POST(url, {
+      data: {
+        title: columnTitle.val(),
+        color: projectColorInput.val(),
+      },
+    });
+
+    if (!response.ok) {
+      throw new Error('Failed to create new column');
```
Comment (delvh) [inline web_src/js/features/repo-projects.js]: Again,
```suggestion
      throw new Error(`Failed to create new column - received status: ${response.status} ${response.statusText}`);
```

File: web_src/js/features/repo-projects.js [hunk 6]
```diff
@@ -67,20 +71,24 @@ async function initRepoProjectSortable() {
...
-          });
+          try {
+            const response = await PUT($(column).data('url'), {
+              data: {
+                sorting: i,
+                color: rgbToHex($(column).css('backgroundColor')),
+              },
+            });
+            if (!response.ok) {
+              throw new Error('Failed to update column sorting');
```
Comment (delvh) [inline web_src/js/features/repo-projects.js]:
```suggestion
              throw new Error(`Failed to update column sorting - received status: ${response.status} ${response.statusText}`);
```

File: web_src/js/features/repo-projects.js [hunk 7]
```diff
@@ -118,26 +126,31 @@ export function initRepoProject() {
...
+      try {
+        const response = await PUT($(this).data('url'), {
+          data: {
+            title: projectTitleInput.val(),
+            color: projectColorInput.val(),
+          },
+        });
+
+        if (!response.ok) {
+          throw new Error('Failed to update project column');
```
Comment (delvh) [inline web_src/js/features/repo-projects.js]:
```suggestion
          throw new Error(`Failed to update project column - received status: ${response.status} ${response.statusText}`);
```

File: web_src/js/features/repo-projects.js [hunk 8]
```diff
@@ -146,19 +159,19 @@ export function initRepoProject() {
...
-        url: $(showButton).data('url'),
-        headers: {
-          'X-Csrf-Token': csrfToken,
-        },
-        contentType: 'application/json',
-      }).done(() => {
+      try {
+        const response = await POST($(showButton).data('url'));
+        if (!response.ok) {
+          throw new Error('Failed to set default project column');
```
Comment (delvh) [inline web_src/js/features/repo-projects.js]:
```suggestion
          throw new Error(`Failed to set default project column - received status: ${response.status} ${response.statusText}`);
```

File: web_src/js/features/repo-projects.js [hunk 9]
```diff
@@ -167,19 +180,19 @@ export function initRepoProject() {
...
-        headers: {
-          'X-Csrf-Token': csrfToken,
-        },
-        contentType: 'application/json',
-        method: 'DELETE',
-      }).done(() => {
+      try {
+        const response = await DELETE(deleteUrl);
+        if (!response.ok) {
+          throw new Error('Failed to delete project column');
```
Comment (delvh) [inline web_src/js/features/repo-projects.js]:
```suggestion
          throw new Error(`Failed to delete project column - received status: ${response.status} ${response.statusText}`);
```

File: web_src/js/features/repo-projects.js (hunk 4 above)
Comment (yardenshoham) [reply web_src/js/features/repo-projects.js]: @silverwind asked for these to be removed so I committed https://github.com/go-gitea/gitea/pull/29814/commits/0520d2f6c5c1b2c87fe0b4abfec9f1da1acf76cd

**Review by delvh — APPROVED** [review]

**Pull request merged**